### PR TITLE
Realize to Dynamics in `createExternalLoadsForGait` +  report `IMU` gyroscope in IMU frame

### PR DESCRIPTION
**Pull request by nickbianco**
Fixes issue #3367, #3385

### Brief summary of changes
- `MocoUtilities::createExternalLoadsForGait` now realizes to `SimTK::Stage::Dynamics` to support all contact models.
- `IMU::calcGyroscopeSignal()` now reports angular velocities in the `IMU` frame.
- Added documentation to `IMU`.

### Testing I've completed
Working on a local test for the `IMU` changes.

### CHANGELOG.md (choose one)

- updated.

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3402)

### CODE REVIEW

**Comment by nickbianco** [comment]
Tested `IMU::calcGyroscopeSignal()` locally, all is working fine.

**Comment by adamkewley** [comment]
LGTM - the only thing I was a bit unsure about was whether `expressVectorInAnotherFrame` also adds the translation (it does not, so it's fine).

File: OpenSim/Simulation/OpenSense/IMU.h [hunk 1]
```diff
@@ -62,21 +66,25 @@ class OSIMSIMULATION_API IMU : public ModelComponent {
...
     }
-    /// Report the orientation of this IMU in ground frame expressed as Quaternion
+    /// Report the orientation of this IMU in ground frame expressed as a
+    /// Quaternion.
     SimTK::Quaternion calcOrientationAsQuaternion(const SimTK::State& s) const {
         return SimTK::Quaternion(calcTransformInGround(s).R());
     }
-    /// Report the angular velocity of the frame to which this IMU is attached
-    /// in ground frame
+    /// Report the angular velocity of this IMU in the frame it is attached to.
```
Comment (aymanhab) [inline OpenSim/Simulation/OpenSense/IMU.h:78]: Should we add a note as to what state realization level is required?

**Review by nickbianco — COMMENTED** [review]
@aymanhab, ready for review.

> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3402#-:-NOaQQ7A5JuaMB0wp-sQ:bfo46z0)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @aymanhab)

___
*[`OpenSim/Simulation/OpenSense/IMU.h` line 78 at r1](https://reviewable.io/reviews/opensim-org/opensim-core/3402#-NOLmMoi7CbL4usfI-G8-r1-78:-NOaQPHk29VvdwMGtvGm:b-896fix) ([raw file](https://github.com/opensim-org/opensim-core/blob/e48ed899773b7bfbe8aaba7ad53b61bdd8964748/OpenSim/Simulation/OpenSense/IMU.h#L78)):*
<details><summary><i>Previously, aymanhab (Ayman Habib) wrote…</i></summary><blockquote>

Should we add a note as to what state realization level is required?
</blockquote></details>

Done.

**Review by aymanhab — APPROVED** [review]
:lgtm:

> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3402#-:-NOa_JCU04hLu2B8ZtRz:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @nickbianco)

**Pull request merged**